pkg/storage/quota_pool.go

```diff
@@ -0,0 +1,81 @@
+package storage
+
+import (
+	"math"
+	"sync"
+
+	"golang.org/x/net/context"
+)
+
+const (
+	leaderProposalQuota   = 1000
+	followerProposalQuota = math.MaxInt64 / 2
+)
+
+type quotaPool struct {
+	c chan int64
+
+	mu    sync.Mutex
+	quota int64
+}
+
+func newQuotaPool(q int64) *quotaPool {
+	qb := &quotaPool{
```

Comment:
Cause I stole it from the `grpc` code base which uses the name `qb`. I'll change it eventually.